### DISCUSSION TITLE
ESP32S3 ADC readings

### DISCUSSION THREAD

**Discussion by saraverbeecke** [micropython/micropython#11883] [ESP32] [answered]
Hi,

Are the MicroPython docs about the ESP32 [ADC](https://docs.micropython.org/en/latest/esp32/quickref.html#adc-analog-to-digital-conversion) also relevant for the ESP32**S3**?

I would like to understand what the read_uv() function means and how to read out a correct analog input voltage.
1. What is the actual measurable input voltage range on an ADC input pin?
2. Do I need to use the attenuation value (the read_uv() is only relevant before the atten)?

Thanks!

**Comment by peterhinch** [comment, marked as answer]
I haven't done detailed tests but I have used the ambient light sensor in the UM Feather S3. The ADC uses the same MicroPython API as the standard ESP32. I needed to use `ADC.ATTN_11DB` to get the right output range.
